extension/goplugin/auth.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2017 NTT Innovation Institute, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goplugin
+
+import (
+	"github.com/cloudwan/gohan/extension/goext"
+	"github.com/cloudwan/gohan/schema"
+)
+
+// Auth is an implementation of IAuth
+type Auth struct{}
+
+// IsAdmin returns if context has admin priviledges
+func (auth *Auth) IsAdmin(context goext.Context) bool {
+	if roleRaw, ok := context["role"]; ok {
+		if role, ok := roleRaw.(*schema.Role); ok {
+			if authRaw, ok := context["auth"]; ok {
+				if auth, ok := authRaw.(schema.Authorization); ok {
+					return role.Match("admin") && auth.TenantName() == "admin"
```

Comment:
As an gohan policy, admin privilege is provided if the role name is "admin". See: https://github.com/cloudwan/gohan/blob/master/etc/schema/gohan.json#L7

Any tenant name is ok.

Reply:
I just copied this code from auth.js:
```
        return role && auth && role.Match("admin") && (auth.TenantName() == "admin")
```
so it means that both JS and Go implementations are incorrect?

Reply:
`auth.js` is our private javascript extension. That means our private function isAdminContext was incorrect...

Actually, I wondered in the previous review whether this `isAdmin` function should be placed in public gohan tree or not. However, in my current opinion, this function itself is ok because public `gohan.json` has a special statement about admin role.

Reply:
I've changed the code to only expose generic functions. Custom logic will be moved to extensions.